jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -151,7 +141,19 @@ else if (urlString.startsWith("jar:"))
      */
     public static Resource newResource(String resource) throws IOException
     {
-        return newResource(resource, __defaultUseCaches);
+        URI uri = URI.create(resource);
+        try
+        {
+            // If the URI has no scheme, we consider the string actually was a path.
+            Path path = uri.getScheme() == null ? Paths.get(resource) : Paths.get(uri);
+            return newResource(path);
```

Comment:
I feels wrong that we are not passing  the created uri into the created resource.  We will almost certainly need it again, so best to avoid recreating it.

Reply:
I've refactored the `PathResource` such as it only accepts a `URI` as its sole constructor argument. Although this suffers from a similar problem problem (`Path.toUri()` is quite expensive too) it feels like the correct thing to do and that we should only optimize this code after we're happy with the behavior.